src/wp-includes/class-wp-block-pattern-categories-registry.php

```diff
@@ -162,30 +162,3 @@ public static function get_instance() {
 		return self::$instance;
 	}
 }
-
-/**
```

Comment:
It's good practice to `require_once` the new location when moving functions. It accounts for theme and plugin devs doing strange things.

https://github.com/WordPress/wordpress-develop/blob/f6c0c517062cf4952cd114ca51972fd8b640cfd3/src/wp-includes/class-wp-block-parser.php#L392-L404

Reply:
This file is coming directly from Gutenberg, that's why I did not change anything here.
Initially I had those removed, but PHPUnit tests were failing (the `Ensure version-controlled files are not modified or deleted during building` test - https://github.com/WordPress/wordpress-develop/actions/runs/7407281586/job/20153188297). So in https://github.com/WordPress/wordpress-develop/pull/3470/commits/a3e6dd3882eca22342270e5cce6b7f7d82caa2dd I reverted that change, and as soon as this gets merged in Core we'll need to tweak this in Gutenberg :+1:

Reply:
The link I provided wasn't great, because it was for classes rather than function movement. Yeah, that should be changed upstream once an autoloader is introduced, sorry for the confusion.

As this PR is moving function definitions, they'll need to include the files the definitions are moved to. This is to account for extenders doing strange things such as custom bootstrapping. Similar to what [Wordfence appear to be doing](https://plugins.trac.wordpress.org/browser/wordfence/tags/7.11.1/waf/bootstrap.php) here via the `auto_prepend_file` option.

Reply:
Specifically, here one would want to replace the now-removed function declarations with this, right?

```php
/** 
 * Block patterns functions.
 * 
 * Required for backward compatibility in WordPress Core. 
 */ 
require_once ABSPATH . WPINC . '/block-patterns.php';
```

(Or maybe `__DIR__ . '/block-patterns.php'`, if you'd want this to be relative.)